pkg/webhook/plugins/plugins_impl.go

```
@@ -0,0 +1,154 @@
+	"github.com/fluid-cloudnative/fluid/pkg/webhook/plugins/mountpropagationinjector"
+	"github.com/fluid-cloudnative/fluid/pkg/webhook/plugins/nodeaffinitywithcache"
+	"github.com/fluid-cloudnative/fluid/pkg/webhook/plugins/prefernodeswithoutcache"
+	"github.com/fluid-cloudnative/fluid/pkg/webhook/plugins/requirenodewithfuse"
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	log = ctrl.Log.WithName("plugins")
+	// registry stores all plugins, mapping by name.
+	registry = api.Registry{}
+
+	// cache handlers to avoid deserialization cost
+	cacheHandlers *Handlers
+
+	fluidNameSpace = common.NamespaceFluidSystem
+)
+
+func init() {
+	nameSpace, err := utils.GetEnvByKey(common.MyPodNamespace)
+	if err != nil || nameSpace == "" {
+		log.Info(fmt.Sprintf("can not get non-empty fluid system namespace from env, use %s", common.NamespaceFluidSystem))
+	} else {
+		fluidNameSpace = nameSpace
+	}
+}
+
+func RegisterMutatingHandlers(client client.Client) {
+	// ignore the register error
+	_ = registry.Register(prefernodeswithoutcache.Name, prefernodeswithoutcache.NewPlugin)
+	_ = registry.Register(mountpropagationinjector.Name, mountpropagationinjector.NewPlugin)
+	_ = registry.Register(requirenodewithfuse.Name, requirenodewithfuse.NewPlugin)
+	_ = registry.Register(nodeaffinitywithcache.Name, nodeaffinitywithcache.NewPlugin)
+	_ = registry.Register(fusesidecar.Name, fusesidecar.NewPlugin)
+
+	// get the handlers through tbe config
+	cm, err := kubeclient.GetConfigmapByName(client, common.PluginProfileConfigMapName, fluidNameSpace)
+
+	// if has error, return empy handlers to avoid nil panic
+	if err != nil {
+		log.Error(err, "get plugins config map error")
+		cacheHandlers = &Handlers{}
+		return
+	}
+	if cm == nil {
+		log.Error(errors.New("plugins config map not exist"), "skip mutating")
+		cacheHandlers = &Handlers{}
+		return
+	}
```

The PR now disables all the plugins by default. Is it possible to enable all the plugins by default if something happened when getting the configmap?

---

Or make it fail early when configmap can't be got?

---

make it fail early when occuring an error.